addons/binding/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/state/ZoneControlState.java

```diff
@@ -25,5 +25,6 @@
     public String inputID = VALUE_EMPTY;
     public String surroundProgram = VALUE_EMPTY;
     public float volume = 0.0f; // volume in percent
+    public float volumeDB = 0.0f; // volume in dB
```

Comment:
The volume should either be stored in db or in percentage, but not in both. Because of the introduced minDB and maxDB I would favour the volumeDB field. The percentage can be computed.

Reply:
Actually both are supported. If the user sets volumeDB the volume percentage is calculated, and vice versa. The actual updates from Yamaha come back in DB if I remember correctly.

Reply:
I agree to have both channels, I just disagree to save both values, especially if one can be computed that easily.

Reply:
I think the way it is right now fits into the existing design. 

Note that in order to move the volume conversion logic out of ZoneControlXML the YamahaZoneConfiguration is also required (for min/max volume DB). Furthermore, if a set volume % command is made then similar conversion needs to be performed to dB in ZoneControlXML. 

Not sure where this logic would sit? The only other option I see is a getter were to be exposed under ZineControlState but then you bring in the config class and break apart the from dB from the to dB conversion.

Let me know what exactly you have in mind.

Reply:
1) remove volume
2) Move the conversion logic into a method YamahaZoneConfiguration.getPercentageVolume(float volumeInDB)->float
3) Use YamahaZoneConfiguration.getPercentageVolume within the handler to return a percentage value

Is that a viable option?
Redundant values are risky for fellow developers who refactor code and forget about keeping those values consistent.

Reply:
Taking aside that the simple config holder is going to be in the business of volume conversion, yes this is viable. Check the change please.